features/04_aruba_api/environment/set_environment_variable.feature

```diff
@@ -12,58 +12,71 @@ Feature: Set environment variable via API-method
   Background:
     Given I use the fixture "cli-app"
 
-  Scenario: Non-existing variable
+  Scenario: Setting a new variable
     Given a file named "spec/environment_spec.rb" with:
     """ruby
     require 'spec_helper'
 
-    RSpec.describe 'Environment command', :type => :aruba do
-      before { set_environment_variable 'LONG_LONG_VARIABLE', '1' }
+    RSpec.describe 'running env', :type => :aruba do
+      before do
+        set_environment_variable 'LONG_LONG_VARIABLE', '1'
 
-      before { run_command('env') }
-      before { stop_all_commands }
+        run_command('env')
+        stop_all_commands
+      end
 
-      it { expect(last_command_started.output).to include 'LONG_LONG_VARIABLE=1' }
+      it "outputs the value set and restores the original" do
+        expect(last_command_started.output).to include 'LONG_LONG_VARIABLE=1'
+        expect(ENV['LONG_LONG_VARIABLE']).to be_nil
+      end
     end
     """
     When I run `rspec`
     Then the specs should all pass
 
-  Scenario: Existing variable set from within the test
+  Scenario: Overriding an previously set variable
     Given a file named "spec/environment_spec.rb" with:
     """ruby
     require 'spec_helper'
 
-    RSpec.describe 'Environment command', :type => :aruba do
-      before { set_environment_variable 'LONG_LONG_VARIABLE', '1' }
-      before { set_environment_variable 'LONG_LONG_VARIABLE', '2' }
+    RSpec.describe 'running env', :type => :aruba do
+      before do
+        set_environment_variable 'LONG_LONG_VARIABLE', '1'
+        set_environment_variable 'LONG_LONG_VARIABLE', '2'
 
-      before { run_command('env') }
-      before { stop_all_commands }
+        run_command('env')
+        stop_all_commands
+      end
 
-      it { expect(last_command_started.output).to include 'LONG_LONG_VARIABLE=2' }
+      it "outputs the last values set" do
+        expect(last_command_started.output).to include 'LONG_LONG_VARIABLE=2'
+      end
     end
     """
     When I run `rspec`
     Then the specs should all pass
 
 
-  Scenario: Existing variable set by some outer parent process
+  Scenario: Setting variable set by some outer parent process
```

Comment:
Does this read right, now? Can it be made shorter and still mean the right thing?

Reply:
I simplified the description.